src/components/chat-item/chat-prompt-input.ts

```diff
@@ -324,6 +324,26 @@ export class ChatPromptInput {
           this.quickPickOpen = true;
         }
       } else if (navigationalKeys.includes(e.key)) {
+        const inputText = this.promptTextInput.getTextInputValue();
+        const cursorPosition = this.promptTextInput.getCursorPos();
+
+        const textBeforeCursor = inputText.substring(0, cursorPosition);
+        const textAfterCursor = inputText.substring(cursorPosition);
+
+        // Check if cursor is on the first line
+        const isFirstLine = !textBeforeCursor.includes('\n');
+
+        // Check if cursor is on the last line
+        const isLastLine = !textAfterCursor.includes('\n');
```

Comment:
just to double check, is it safe to deduce first/last line by checking for `\n`? i am thinking of some edge cases with input copy pasted starting/ending with empty lines

Reply:
In fact, I tried with starting/ending the input with some empty lines (shift + return), its quite a rare edge case i guess though

Reply:
I did some more testing, and I think it is not a safe deduction to check for `'\n'` to deduce first/last line. To do this we might need to check the height of the input field or count the n of lines somehow

Reply:
Agree with Francesco.